kratos/geometries/line_3d_2.h

```diff
@@ -403,6 +403,26 @@ class Line3D2 : public Geometry<TPointType>
     }
 
 
+    ///@}
+    ///@name Intersections
+    ///@{
+
+    /**
+     * @brief Test if this geometry intersects with other geometry
+     * @param  ThisGeometry Geometry to intersect with
+     * @return              True if the geometries intersect, False in any other case.
+     */
+    bool HasIntersection(const BaseType& rThisGeometry) const override {
+        const BaseType& geom = *this;
+        if (rThisGeometry.WorkingSpaceDimension() > geom.WorkingSpaceDimension()) {
+            return rThisGeometry.HasIntersection(geom);
+        }
+        // Both objects are lines
+        Point intersection_point;
+        return IntersectionUtilities::ComputeLineLineIntersection(geom, rThisGeometry[0], rThisGeometry[1], intersection_point);
```

Comment:
Maybe I'm missing something but, if I call this method with a 3D triangle, whose working space dimension is 2 as well, we will reach this point right?

Reply:
OK, I think that this is fixed by [this](https://github.com/KratosMultiphysics/Kratos/pull/9360#discussion_r752085976).

Reply:
yes, it has been fixed with [this](https://github.com/KratosMultiphysics/Kratos/pull/9360#discussion_r752085976)